src/Microsoft.Health.Functions.Extensions/AzureFunctionsJobHostBuilder.cs

```diff
@@ -0,0 +1,133 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using EnsureThat;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Memory;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Functions.Extensions.Configuration;
+
+namespace Microsoft.Health.Functions.Extensions;
+
+/// <summary>
+/// Represents a builder for instances of <see cref="IHost"/> that run Azure Functions.
+/// </summary>
+public sealed class AzureFunctionsJobHostBuilder
```

Comment:
This type is meant to be used when running integration tests. Check out the `WebJobsTestFixture` for an E2E example of its usage